examples/python/interface.py

```diff
@@ -96,150 +72,189 @@ def control_test(testcase_name, control_module='', start_time=0, warmup_period=0
-    # Instantiate concrete controller (pid, pidTwoZones, sup, etc.)
-    controller = Controller(control_module, use_forecast)
+    def check_response(response):
+        """Check status code from restful API and return result if no error
+        Shut down test case if error in API call
 
-    # GET TEST INFORMATION
-    # -------------------------------------------------------------------------
-    print('\nTEST CASE INFORMATION\n---------------------')
-    # Select test case
-    testid = requests.post("{0}/testcases/{1}/select".format(url, testcase_name)).json()["testid"]
-    print('TestID:\t\t\t\t{0}'.format(testid))
-    # Retrieve testcase name from REST API
-    name = check_response(requests.get('{0}/name/{1}'.format(url,testid)))
-    print('Name:\t\t\t\t{0}'.format(name))
-    # Retrieve a list of inputs (controllable points) for the model from REST API
-    inputs = check_response(requests.get('{0}/inputs/{1}'.format(url,testid)))
-    print('Control Inputs:\t\t\t{0}'.format(inputs))
-    # Retrieve a list of measurements (outputs) for the model from REST API
-    measurements = check_response(requests.get('{0}/measurements/{1}'.format(url,testid)))
-    print('Measurements:\t\t\t{0}'.format(measurements))
-
-    # IF ANY CUSTOM KPI CALCULATION, DEFINE STRUCTURES
-    # ------------------------------------------------
-    custom_kpis = []  # Initialize customized kpi calculation list
-    custom_kpi_result = {}  # Initialize tracking of customized kpi calculation results
-    if customized_kpi_config is not None:
-        with open(customized_kpi_config) as f:
-            config = json.load(f, object_pairs_hook=collections.OrderedDict)
-        for key in config.keys():
-            custom_kpis.append(CustomKPI(config[key]))
-            custom_kpi_result[CustomKPI(config[key]).name] = []
-    custom_kpi_result['time'] = []
-
-    # RUN TEST CASE
-    # -------------------------------------------------------------------------
-    # Record real starting time
-    start = time.time()
-    # Initialize test case
-    print('Initializing test case simulation.')
-    # Check if a scenario is defined
-    if scenario is not None:
-        # Initialize test with a scenario and get response for time_period field
-        res = check_response(requests.put('{0}/scenario/{1}'.format(url,testid), json=scenario))['time_period']
-        if res == None:
-            # If no time_period was specified (only electricity_price), initialize test with a specified start time and warmup period
+        Parameters
+        ----------
+
+        response: obj, response object
+
+        Returns
+        -------
+        result : dict, result from call to restful API
+
+        """
+        if isinstance(response, requests.Response):
+            status = response.status_code
+        if status == 200:
+            response = response.json()['payload']
+            return response
+        print("Unexpected error: {}".format(response.text))
```

Comment:
Suggest also reporting status code since in BOPTEST we do also return 400 or 500 depending on the error.